src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -387,4 +401,26 @@ TEST_CASE("request_rate_sequence")
   trrm.TestSequences();
 }
 
+TEST_CASE("request_rate_streaming: test that streaming-specific logic works")
+{
+  PerfAnalyzerParameters params{};
+  params.streaming = true;
+
+  TestRequestRateManager trrm(params);
+  trrm.schedule_.push_back(std::chrono::nanoseconds(1));
+
+  std::shared_ptr<TestRequestRateManager::ThreadStat> thread_stat{
+      std::make_shared<TestRequestRateManager::ThreadStat>()};
```

Comment:
Why not use the `ThreadStat` and `ThreadConfig` objects from the RequestRateManager itself?

Reply:
They're private and inaccessible from the test class. I'm simply "revealing a reference" to the private classes by creating a new struct that inherits directly from the private one.